merlin/models/tf/blocks/core/combinators.py

```diff
@@ -602,3 +603,69 @@ def __init__(
             strict=strict,
             **kwargs,
         )
+
+
+@tf.keras.utils.register_keras_serializable(package="merlin.models")
+class Cond(Layer):
+    """Layer to enable conditionally apply layers."""
+
+    def __init__(self, condition: Layer, true: Layer, false: Optional[Layer] = None, **kwargs):
+        super(Cond, self).__init__(**kwargs)
+        self.condition = condition
+        self.true = true
+        self.false = false
+
+    def call(self, inputs, **kwargs):
+        """Call layers conditionally."""
+        condition = call_layer(self.condition, inputs, **kwargs)
+
+        def true_fn():
+            return call_layer(self.true, inputs, **kwargs)
+
+        def false_fn():
+            if self.false is None:
+                return inputs
+            return call_layer(self.false, inputs, **kwargs)
+
+        return tf.cond(tf.convert_to_tensor(condition), true_fn, false_fn)
+
+    def compute_output_shape(self, input_shape):
+        """Computes the output shape of the layer."""
+        true_output_shape = self.true.compute_output_shape(input_shape)
+
+        if self.false:
+            false_output_shape = self.false.compute_output_shape(input_shape)
+        else:
+            false_output_shape = input_shape
+
+        if true_output_shape != false_output_shape:
```

Comment:
Should we add a flag in `init` to specify whether this should throw an exception?